### PR TITLE
[FIXED JENKINS-23995] Set context for GH status

### PR DESCRIPTION
**Pull request by KostyaSha**
supersede #47

### CODE REVIEW

**Comment by KostyaSha** [comment]
@janpapenbrock please test this hpi (press green checkbox and then click on "module builds")

**Comment by janpapenbrock** [comment]
Thanks very much for your effort. Still have a similar result, but the names have changed: https://raw.githubusercontent.com/janpapenbrock/github-plugin-test/master/3-notifications.png

While we are at it, I have a probem with getting the SHA1 to be set correctly in downstream builds - I have to clone it again although it was cloned upstream. This is probably a different issue, but its quite near in the code I guess. Should be this https://github.com/KostyaSha/github-plugin/blob/JENKINS-23995/src/main/java/com/cloudbees/jenkins/GitHubCommitNotifier.java#L102

Maybe this is related to the problem seen here?!?

**Comment by KostyaSha** [comment]
@janpapenbrock have you added new commit to branch where you testing statuses?

**Comment by janpapenbrock** [comment]
Yes, I did.

**Comment by janpapenbrock** [comment]
The Github API documentation says at https://developer.github.com/v3/repos/statuses/#create-a-status:

context 	string 	A string label to differentiate this status from the status of other systems. Default: "default"

So maybe this issue is related to the API implementation?

**Comment by KostyaSha** [comment]
@janpapenbrock stop guessing ;) 
Stases is set multiple time, i trusted initial PR code. Now should work right. Wait PR build results and test new hpi.

**Comment by janpapenbrock** [comment]
Great thanks ;)

**Comment by KostyaSha** [comment]
PR builds fixed, new hpi is available

**Comment by janpapenbrock** [comment]
Success! :)

https://github.com/janpapenbrock/github-plugin-test/blob/master/4-success.png

Thank you!

**Comment by KostyaSha** [comment]
Cool, need +1 from some other dev and will merge.

File: src/main/java/com/cloudbees/jenkins/GitHubSetCommitStatusBuilder.java [hunk 1]
```diff
@@ -29,7 +29,7 @@ public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListen
         for (GitHubRepositoryName name : GitHubRepositoryNameContributor.parseAssociatedNames(build.getProject())) {
             for (GHRepository repository : name.resolve()) {
                 listener.getLogger().println(Messages.GitHubCommitNotifier_SettingCommitStatus(repository.getUrl() + "/commit/" + sha1));
-                repository.createCommitStatus(sha1, GHCommitState.PENDING, build.getAbsoluteUrl(), Messages.CommitNotifier_Pending(build.getDisplayName()));
+                repository.createCommitStatus(sha1, GHCommitState.PENDING, build.getAbsoluteUrl(), Messages.CommitNotifier_Pending(build.getProject().getFullName()), build.getProject().getFullName());
```
Comment (oleg-nenashev) [inline src/main/java/com/cloudbees/jenkins/GitHubSetCommitStatusBuilder.java]: Any reasons to change the <code>DisplayName</code> to the project's full name. In such cases the status info will lose the information about the build.

Comment (KostyaSha) [inline src/main/java/com/cloudbees/jenkins/GitHubSetCommitStatusBuilder.java]: No, where this info displayed?

Comment (KostyaSha) [inline src/main/java/com/cloudbees/jenkins/GitHubSetCommitStatusBuilder.java]: Fixed and tested, result is here https://github.com/KostyaSha/test-repo/pull/20

**Comment by KostyaSha** [comment]
@kohsuke @oleg-nenashev i will merge and release this state if i will get no replies in the next day.

**Pull request merged**